### PR TITLE
design: Proxy status design doc

### PR DESCRIPTION
**Pull request by stevesloka**
Updates #1598 

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

File: design/proxy-status.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+### Add fields to the Status object
+
+Currently the `status` field of an `HTTPProxy` contains two fields:
+
+- **Status**: States a quick one-word status of the object (i.e. `Valid` / `Invalid`)
+- **Description**: States a more detailed explanation of the current status
+
+The Status object could be expanded to also hold a set of merged `Conditions`. When a user creates an HTTPProxy, the status field will be updated with this information regardless of the state of the object (i.e. invalid or valid).
+
+```yaml
```
[the remark this answers is not in the document]
Comment (stevesloka) [reply design/proxy-status.md]: It doesn't matter where it comes from or how many proxies include this child, the idea is that the end result will be shown. So for a path, if there were 4 different includes that resulted in `/foo/bar/zed/api` then that's what's shown. I'm also not sure we should leak the namespace / root info out to users.

File: design/proxy-status.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  Description:     valid HTTPProxy
+  Included Conditions:
+     prefix: /marketing/blog
+     headerContains:
+       x-header: abc
+     headerExact:
+       another-header: somethingelse
+```
+
+Addionally, HTTPProxies could include a "dry-run" field or possibly add a second CRD type (e.g. `httpproxy-dryrun`). With this capability, users could first see what conditions they were Included, understand if their proxy would be valid (although would probably be orphan type).
```
[the remark this answers is not in the document]
Comment (stevesloka) [reply design/proxy-status.md]: What coordination is needed between admin/user?

File: design/proxy-status.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+       - prefix: /marketing/blog
+       - headerContains:
+           x-header: abc
+       - headerExact:
+           another-header: somethingelse
+```
+
+####  Issues
+
+The problem with this design is that Contour needs write access to any namespace that contains Proxies which could pose as a security risk.
```
[the remark this answers is not in the document]
Comment (stevesloka) [reply design/proxy-status.md]: Ahh good point! 👍

File: design/proxy-status.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (youngnick) [reply design/proxy-status.md]: The admin and user must agree on both the namespace and name of the object to be included, and the admin should make the user aware of the conditions on the inclusion. If the first is not true, the user has no way to know what to call their thing. If the admin does not that, a full listing of the include path is necessary, for the user to be able to either fix their HTTPProxy, or to tell the admin that the behaviour is not desired.

**Pull request closed without merging**